src/Compiler/Microsoft.CodeAnalysis.Razor/src/GenericTypeNameRewriter.cs

```diff
@@ -49,7 +49,7 @@ public override SyntaxNode Visit(SyntaxNode node)
                     //
                     // We add our own diagnostics for missing/invalid type parameters anyway.
                     var replacement = binding == null ? typeof(object).FullName : binding;
-                    return identifier.Update(SyntaxFactory.Identifier(replacement));
+                    return identifier.Update(SyntaxFactory.Identifier(replacement).WithTriviaFrom(identifier.Identifier));
```

Comment:
Reusing trivia will preserve comments correct? Is that intended? Is it okay if we end up preserving a multi-line comment between the two files?

Reply:
> Reusing trivia will preserve comments correct?

Yes, comments are preserved.

> Is that intended?

FWIW, it's consistent with `GlobalQualifiedTypeNameRewriter` (from fix #6805):

https://github.com/dotnet/razor/blob/c2d6c0769ea4bfe8639f6a4125744581419139c7/src/Compiler/Microsoft.CodeAnalysis.Razor/src/GlobalQualifiedTypeNameRewriter.cs#L63

> Is it okay if we end up preserving a multi-line comment between the two files?

It doesn't seem to break anything. If it's OK in a generic tuple `(T1 X, T2 Y)`, it will be OK in the specific one, e.g., `(int X, int Y)`.

Reply:
Can we add some tests with multiline comments here? In particular, scenarios that look something like this:

```cs
List<(TItem1 X // Test
, TItem2 Y)>
```

```cs
List<(TItem1 X /* Test
another line */,
TItem Y)>
```

And other such variations.

Reply:
Added some unit test data with comments.